domain/agentbinary/service/store_test.go

```diff
@@ -228,8 +271,42 @@ func (s *storeSuite) TestAddWithSHA256(c *gc.C) {
 			Number: semversion.MustParse("4.6.8"),
 			Arch:   corearch.AMD64,
 		},
-		1234,
-		"test-sha256",
+		size,
+		sha256Hash,
 	)
 	c.Assert(err, jc.ErrorIsNil)
 }
+
+func (s *storeSuite) TestAddWithSHA256FailedInvalidSHA(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	agentBinary := strings.NewReader("test-agent-binary")
+
+	store := NewAgentBinaryStore(s.mockState, loggertesting.WrapCheckLog(c), s.mockObjectStoreGetter)
+	err := store.AddWithSHA256(context.Background(), agentBinary,
+		coreagentbinary.Version{
+			Number: semversion.MustParse("4.6.8"),
+			Arch:   corearch.AMD64,
+		},
+		1234,
+		"invalid-sha256",
+	)
+	c.Assert(err, jc.ErrorIs, coreerrors.NotValid)
+}
+
+func (s *storeSuite) TestAddWithSHA256FailedInvalidAgentVersion(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+
+	agentBinary := strings.NewReader("test-agent-binary")
+	sha256Hash, _ := s.calculateSHA(c, "test-agent-binary")
+
+	store := NewAgentBinaryStore(s.mockState, loggertesting.WrapCheckLog(c), s.mockObjectStoreGetter)
+	err := store.AddWithSHA256(context.Background(), agentBinary,
+		coreagentbinary.Version{
+			Arch: corearch.AMD64,
```

Comment:
The version package has a `Zero` member that we can assign here to get the same result. Just reads a little more explicit in my opinion. Your call on this.